### PR TITLE
fix: revert module change

### PR DESCRIPTION
**Pull request by vacas5**
This steps back a complete shift from CommonJS to esm modules, and instead makes only the file that imports tablemark use esm. Then, we use a dynamic import statement inside the build script to import that file as a workaround. Thus we're able to leave tablemark as is for now.

Closes D2IQ-92904

# Description

## Checklist

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [ ] This PR is associated with a JIRA and mentions in the commit message footer ("Closes …")
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [ ] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -17,9 +17,7 @@
     "skipLibCheck": true,
     "tsBuildInfoFile": ".tsbuildinfo",
     "lib": ["es6", "dom", "es2017"],
-    "incremental": true,
-    "declaration": true,
-    "emitDeclarationOnly": true // to emit at least something until we can update Typescript to v4 then we will want to use `--noEmit`
```
Comment (vacas5) [inline tsconfig.json:22]: I moved these back out because the linter used this too and I didn't want any adverse side effects with it, either

Comment (mstruebing) [reply tsconfig.json:22]: 💬 Feels weird that we need an own configuration file for type checks 🤷

File: packages/design-tokens/scripts/build.js [hunk 2]
```diff
@@ -17,18 +15,20 @@ const DOCS_CONFIG = "./packages/design-tokens/config.designGuidelines.json";
 const configs =
   formatCategory === "docs" ? [DOCS_CONFIG] : [CONFIG, DIST_CONFIG];
 
-configs.forEach(path => {
-  const styleDictionary = StyleDictionary.extend(path);
+import("./formats.mjs").then(({ default: formats }) => {
```
Comment (mstruebing) [inline packages/design-tokens/scripts/build.js:18]: Is that the magic 🪄 ?

**Review by mstruebing — APPROVED** [review]
It seems to work fine - thanks for doing this 🥇

File: tsconfig.json (hunk 1 above)
Comment (vacas5) [reply tsconfig.json:22]: I think it's the github action that needed `emitDeclarationOnly`. So, we need a config file in order to do type checking with a github action, maybe?

File: packages/design-tokens/scripts/build.js (hunk 2 above)
Comment (vacas5) [reply packages/design-tokens/scripts/build.js:18]: yep, this is it. The trick is that it is a Promise, so I had to find the right place to do it. I tried so many other things.

**Pull request merged**